blockchain/chain/src/store/index.rs

```diff
@@ -54,25 +53,45 @@ pub(super) mod checkpoint_tipsets {
     // Represents a static map of validated tipset hashes which helps to remove the need to validate the tipset
     // back to genesis if it has been validated before, thereby reducing boot times.
-    pub(super) static TIPSET_CHECKPOINTS: Lazy<TipsetCheckpointsRegistry> = Lazy::new(|| {
+    // NB: Add desired tipset checkpoints below this by using RPC command: forest-cli chain tipset-hash <cid keys>
+    // and one can use forest-cli chain validate-tipset-checkpoints to validate tipset hashes
+    // for entries that fall within the range of epochs in current downloaded snapshot file.
+    type TipsetCheckpointsRegistry = HashMap<TipsetKeyHash, GenesisTipsetCids>;
+    pub static CALIBNET_CHECKPOINTS: Lazy<TipsetCheckpointsRegistry> = Lazy::new(|| {
         let mut map = HashMap::new();
-        // The numeric suffix in the variable name is the height of the snapshot.
-        // NB: Add desired tipset checkpoints below this by using RPC command: forest-cli chain tipset-hash <cid keys>
-        const CALIBNET_CHECKPOINT_1405400: TipsetKeyHash = "7930ad8bf32b35314b3bc47b9e25249af8ec6ba7f5544c05e8b5bd3b3ec09f76df8bd2278f9b318badf1a08d0a468abd55130465c6c55f99e67badc0e614ca79";
-        const MAINNET_CHECKPOINT_2325300: TipsetKeyHash = "319f2351ceaf78fbcc8688dc75a19bdf8ee6e895e547ff5cc2f7b18a3a36b65ff94c1860733137d0244352f82ba6fd9672aec14deee358e7cf6e088bf89a28b1";
         const CALIBNET_CHECKPOINT_41000: TipsetKeyHash = "1a11a07d427348cc14eaa901de1ba9c6a4e18400bb557f5a0fabbcb22352319e31a7bc988a92525339f84275c0ef6dfbffcb50bb9d9843701875eecfa3ccb069";
-        add_calibnet!(map, CALIBNET_CHECKPOINT_1405400);
         add_calibnet!(map, CALIBNET_CHECKPOINT_41000);
+        map
+    });
+
+    pub static MAINNET_CHECKPOINTS: Lazy<TipsetCheckpointsRegistry> = Lazy::new(|| {
+        let mut map = HashMap::new();
+        const MAINNET_CHECKPOINT_2325300: TipsetKeyHash = "319f2351ceaf78fbcc8688dc75a19bdf8ee6e895e547ff5cc2f7b18a3a36b65ff94c1860733137d0244352f82ba6fd9672aec14deee358e7cf6e088bf89a28b1";
         add_mainnet!(map, MAINNET_CHECKPOINT_2325300);
         map
     });
 
     pub(super) fn genesis_from_checkpoint_tipset(tsk: &TipsetKeys) -> Option<GenesisTipsetCids> {
-        TIPSET_CHECKPOINTS.get(&tipset_hash(tsk).as_str()).cloned()
+        MAINNET_CHECKPOINTS
+            .get(tipset_hash(tsk).as_str())
+            .or_else(|| CALIBNET_CHECKPOINTS.get(tipset_hash(tsk).as_str()))
+            .cloned()
+    }
+
+    pub fn get_tipset_hashes(network: &str) -> HashSet<String> {
+        match network {
+            "mainnet" => {
+                let keys = MAINNET_CHECKPOINTS.keys().map(|s| s.to_string());
+                HashSet::from_iter(keys)
+            }
+            "calibnet" => {
+                let keys = CALIBNET_CHECKPOINTS.keys().map(|s| s.to_string());
+                HashSet::from_iter(keys)
+            }
+            _ => unreachable!("invalid network id"),
```

Comment:
This is not unreachable. There are other networks than just calibnet and mainnet. We don't have any checkpoints for those networks yet so there's nothing to validate.

Reply:
Fixed, returns a None